src/backend/optimizer/plan/setrefs.c

```diff
@@ -1081,12 +1080,7 @@ set_plan_refs(PlannerInfo *root, Plan *plan, int rtoffset)
 					splan->returningLists = newRL;
 
 					/*
-					 * Set up the visible plan targetlist as being the same as
-					 * the first RETURNING list. This is for the use of
-					 * EXPLAIN; the executor won't pay any attention to the
-					 * targetlist.  We postpone this step until here so that
-					 * we don't have to do set_returning_clause_references()
-					 * twice on identical targetlists.
+					 * copy the new returningList as it may be needed later.
 					 */
 					splan->plan.targetlist = copyObject(linitial(newRL));
```

Comment:
> can't other nodes refer to it?

In other words, Material node uses the original returningList, that was copied to ModifyTable targetList inside make_modifytable. Then,  inside the set_plan_refs,  ModifyTable replaces its targetList and returningList by a newRL. From that point Material and ModifyTable may reference different targetLists. Can it be potentially be harmful?

Reply:
As I understand it, the `set_returning_clause_references` function only changes `varna` and this comment says that this is a normal situation that they may differ in Material and outer plan. https://github.com/arenadata/gpdb/blob/23babbbd5abfe7098a00e7ee72231a0aeef257d0/src/backend/executor/nodeMaterial.c#L352-L358

Reply:
Yes, execution is not affected. But for EXPLAIN output can there be a problem? When `set_returning_clause_references` changes the entries, and Material does not?

Reply:
EXPLAIN uses only attribute names. which should not change.